libs/core/src/lib/utils/directives/truncate/truncate.directive.ts

```diff
@@ -84,12 +112,32 @@ export class TruncateDirective implements OnChanges, AfterViewInit, OnDestroy {
         return this._elementRef.nativeElement;
     }
 
+    /** @hidden
+     * Sets fdTruncateTargetInnerText
+     */
+    _setTextContent(): void {
+        this._fdTruncateTargetTextContent = this._elementRef.nativeElement.textContent;
+    }
+
+    /** @hidden
+     * Sets default style
+     */
+    _setDefaultStyle(): void {
+        this._defaultStyle = this._elementRef.nativeElement.style.cssText;
+    }
+
     /** @hidden */
     ngAfterViewInit(): void {
         if (this.rootElement) {
+            this._setTextContent();
+            this._setDefaultStyle();
+
+            this.refreshTarget(this);
+            this.reset();
+
             this._initTruncate();
 
-            this.windowResize$ = fromEvent(window, 'resize')
+            this._windowResize$ = fromEvent(window, 'resize')
```

Comment:
instead of this, we could use `resizeObservable` from the `@fundamental-ngx/core/utils`.